### PR TITLE
Suppress preview .NET message when building dev branch

### PR DESCRIPTION
**Pull request by zivkan**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/2593

Regression? No

## Description

PR title and diff are quite self explanatory

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: build/common.project.props [hunk 1]
```diff
@@ -86,6 +86,11 @@
     <LangVersion Condition="'$(DotNetBuildFromSource)' == 'true'">latest</LangVersion>
   </PropertyGroup>
 
+  <!-- NuGet's dev branch always targets the next upcoming version of .NET, so we always use the preview SDK -->
+  <PropertyGroup Condition=" '$(IsEscrow)' != 'true' ">
+    <SuppressNETCoreSdkPreviewMessage>true</SuppressNETCoreSdkPreviewMessage>
```
Comment (zivkan) [inline build/common.project.props:91]: After creating this PR, I had a thought that I could just set this as an environment variable on my machine. If we want to suppress it on CI, we can also use an environment variable there.  Is that better/worse than setting the MSBuild property?

Comment (nkolev92) [reply build/common.project.props:91]: I do the env var locally trick, I don't mind suppressing it here.

**Review by nkolev92 — APPROVED** [review]

**Review by jgonz120 — APPROVED** [review]

File: build/common.project.props [hunk 2]
```diff
@@ -86,6 +86,11 @@
     <LangVersion Condition="'$(DotNetBuildFromSource)' == 'true'">latest</LangVersion>
   </PropertyGroup>
 
+  <!-- NuGet's dev branch always targets the next upcoming version of .NET, so we always use the preview SDK -->
+  <PropertyGroup Condition=" '$(IsEscrow)' != 'true' ">
```
Comment (jgonz120) [inline build/common.project.props:90]: What does IsEscrow mean?

Comment (zivkan) [reply build/common.project.props:90]: Our VS insertion pipeline uses it to figure out if it needs to insert into the VS main branch, or a release branch. Therefore it's also a way to figure out if a specific commit is likely in our dev branch or a release branch.

**Pull request merged**